planner/core/preprocess.go

```diff
@@ -1610,7 +1611,8 @@ func (p *preprocessor) handleAsOfAndReadTS(node *ast.AsOfClause) {
 //    - session variable
 //    - transaction context
 func (p *preprocessor) ensureInfoSchema() infoschema.InfoSchema {
```

Comment:
Will the tests for stale read pass after this modification?

Reply:
of course. but it seems some problem in CI now.

Reply:
Please check the output of the below sql:

```
set @@tidb_enable_noop_functions=1;
create temporary table test.t1(id int);

create table test.ta (id int primary key);
insert into test.ta values(1);
set @a=now(6);do sleep(1);
drop table test.ta;
create table test.ta (id int primary key);
select * from test.ta as of timestamp(@a) where id=1;

```
The right result will output one line with value "1".

Reply:
Got your meaning, I will fix it in better way.